test/integration/kubernetes_deploy_test.rb

```diff
@@ -1,27 +1,134 @@
 require 'test_helper'
 
 class KubernetesDeployTest < KubernetesDeploy::IntegrationTest
-  def test_basic
-    runner = KubernetesDeploy::Runner.new(
-      namespace: @namespace,
-      current_sha: "abcabcabc",
-      context: MINIKUBE_CONTEXT,
-      template_dir: File.expand_path("./test/fixtures/basic"),
-      wait_for_completion: true,
-    )
-    runner.run
-
-    pods = kubeclient.get_pods(namespace: @namespace)
-    managed_pods, unmanaged_pods = pods.partition do |pod|
-      pod.metadata.ownerReferences && pod.metadata.ownerReferences.first.kind == "ReplicaSet"
+  def test_full_basic_set_deploy_succeeds
+    deploy_fixture_set("basic")
+    basic = FixtureSetAssertions::Basic.new(@namespace)
+    basic.assert_all_up
+  end
+
+  def test_partial_deploy_followed_by_full_deploy
+    deploy_fixture_set("basic", subset: ["configmap-data", "redis"])
+    basic = FixtureSetAssertions::Basic.new(@namespace)
+    basic.assert_all_redis_resources_up
+    basic.assert_configmap_data_present
+    basic.refute_managed_pod_exists
+    basic.refute_web_resources_exist
+
+    deploy_fixture_set("basic")
+    basic.assert_all_up
+  end
+
+  def test_pruning
+    deploy_fixture_set("basic")
+    basic = FixtureSetAssertions::Basic.new(@namespace)
+    basic.assert_all_up
+
+    deploy_fixture_set("basic", subset: ["redis"])
+    basic.assert_all_redis_resources_up
+    basic.refute_configmap_data_exists
+    basic.refute_managed_pod_exists
+    basic.refute_web_resources_exist
+  end
+
+  def test_pvcs_are_not_pruned
+    deploy_fixture_set("basic", subset: ["redis"])
+    basic = FixtureSetAssertions::Basic.new(@namespace)
+    basic.assert_all_redis_resources_up
+
+    deploy_fixture_set("basic", subset: ["configmap-data"])
+    basic.assert_configmap_data_present
+    basic.refute_redis_resources_exist(expect_pvc: true)
+  end
+
+  def test_success_with_unrecognized_resource_type
+    # Secrets are intentionally unsupported because they should not be committed to your repo
+    fixture_set = load_fixture_data("basic", ["configmap-data"])
+    secret = {
+      "apiVersion" => "v1",
+      "kind" => "Secret",
+      "metadata" => { "name" => "test" },
+      "data" => { "foo" => "YmFy" }
+    }
+    fixture_set["secret"] = { "Secret" => secret }
+    deploy_loaded_fixture_set(fixture_set)
+
+    live_secret = kubeclient.get_secret("test", @namespace)
+    assert_equal({ foo: "YmFy" }, live_secret["data"].to_h)
+  end
+
+  def test_invalid_yaml_fails_fast
+    error = assert_raises(KubernetesDeploy::FatalDeploymentError) do
+      deploy_fixture_set("invalid", subset: ["yaml-error"])
+    end
+    assert_match /Template \S+ cannot be parsed/, error.to_s
```

Comment:
I think you can use `assert_raises` with a regex as the second argument here to assert on the message.

```ruby
error_message = "/Template \S+ cannot be parsed/"
assert_raises(KubernetesDeploy::FatalDeploymentError, error_message) { .. }
```